### PR TITLE
Undo button

### PR DESCRIPTION
**Pull request by chrisjpurdy**
If we want to have a general "undo anything" button, there's a weird edge case with `PromotableValuePresenter` (I think) - when "Undo" is pressed, the presenter doesn't update its displayed value, but does if you force it to re-render another way. 

Because of this annoying edge case, I've made it so that actions are only recorded as "undoable" at the level of whatever calls `update`, i.e. the accordion section deletion function passes a flag to `update` saying "allow this action to be undone".
As a rule of thumb structural changes (deleting or moving tabs, accordion sections, content blocks etc.) can be undone without any problems (as far as I can tell).

Currently only deletions are marked as invertible, and only one "last action" is recorded at a time.

I've added a flash to the undo button when it appears. Adding a toast the first time you delete something pointing out that the button exists might be nice, but I didn't feel it was necessary? Open to thoughts on this!

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -6247,6 +6252,11 @@ jest@^27.4.3:
     import-local "^3.0.2"
     jest-cli "^27.5.1"
 
+jiff@^0.7.3:
```
Comment (mwtrew) [inline yarn.lock:6255]: We don't appear to use this, so it's probably worth getting rid of by re-generating `yarn.lock`.

File: src/utils/inversePatch.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    test: invertTest,
+    add: invertAdd,
+    replace: invertReplace,
+    remove: invertRemove,
+    move: invertMove,
+    copy: invertCopy
+};
+
+export function inverseOperation(p: Operation[]) {
+    let pr: Operation[] = [];
```
Comment (mwtrew) [inline src/utils/inversePatch.ts:22]: I realise the code this is based on looks like this too, but I find this unnecessarily hard to follow for lack of useful variable names.

File: src/styles/editor.module.css [hunk 3]
```diff
@@ -207,6 +207,18 @@
     background: #eee;
 }
 
+@keyframes flash {
```
Comment (mwtrew) [inline src/styles/editor.module.css:210]: This didn't draw my attention to the button appearing, it could be made less subtle!

**Review by mwtrew — CHANGES_REQUESTED** [review]
Seems to work well and consistently across different content blocks, which is nice. I have made a couple of minor suggestions, the only functional one being that I didn't notice the button appearing.

File: src/utils/inversePatch.ts (hunk 2 above)
Comment (chrisjpurdy) [reply src/utils/inversePatch.ts:22]: Completely agree, I didn't really want to change this code at all but maybe it would be best since it's _fairly_ short

File: src/styles/editor.module.css (hunk 3 above)
Comment (chrisjpurdy) [reply src/styles/editor.module.css:210]: To be honest, I can't think of how to improve this without it being overly obnoxious. I'm sure once the content team know it's there they won't need it to be highlighted, so maybe this flash ends up being more cosmetic?

**Review by mwtrew — APPROVED** [review]
